### ISSUE TITLE
Can position cursor in "between" single emoji character

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#13404]
Flag emojis are currently not recognized as one character. They are internally represented as multiple characters and with a11y means (via volume key cursor movement) the cursor can be positioned between those two internal chars, which is nonsense.

**Comment by zoechi** [comment]
I still need press the arrow key 4x to get the cursor past the flag character.

```
[✓] Flutter (Channel master, v1.2.2-pre.21, on Mac OS X 10.14.3 18D42, locale en-AT)
[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)
```

**Comment by justinmc** [comment]
Here's a partial solution: https://github.com/flutter/engine/pull/17420

I think it will only fix this for Dpad movement, but we might be able to take a similar approach to solve it for other ways to move the caret.

**Comment by justinmc** [comment]
This should solve it for using the arrow keys, etc. to put the cursor in between emojis: https://github.com/flutter/flutter/pull/53381

**Comment by markusaksli-nc** [comment]
Reproducible on the latest master `1.21.0-2.0.pre.114`.
Touch and Samsung keyboard drag are fixed. Arrow keys are fixed on web and Pixel C tablet emulator. With volume keys cursor control on android, it is still possible to scroll through the emoji as separate characters. On web it is possible to click and place the cursor in the middle of the emoji.
<details>
 <summary>Minimal reproducible code sample</summary>

 ```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Sample',
      home: Scaffold(
        appBar: AppBar(
          title: Text('Sample'),
        ),
        body: Container(
          child: Center(
            child: TextFormField(initialValue: "   👨‍👩‍👦   ",),
          ),
        ),
      ),
    );
  }
}
```
</details>
<details>
 <summary>flutter doctor -v</summary>

 ```
[√] Flutter (Channel master, 1.21.0-2.0.pre.114, on Microsoft Windows [Version 10.0.18362.900], locale et-EE)
    • Flutter version 1.21.0-2.0.pre.114 at C:\Development\flutter_master
    • Framework revision ddb8e6e3bf (7 hours ago), 2020-07-22 20:00:07 -0700
    • Engine revision dcc9a4048d
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 9dca49e71e)


[√] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at C:\Users\Isa\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: C:\Users\Isa\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.6.30309.148
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.0)
    • Android Studio at C:\Users\Isa\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.47.2)
    • VS Code at C:\Users\Isa\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.12.2

[√] Connected device (5 available)
    • SM G950F (mobile)       • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • sdk gphone x86 (mobile) • emulator-5554      • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows            • windows-x64    • Microsoft Windows [Version 10.0.18362.900]
    • Web Server (web)        • web-server         • web-javascript • Flutter Tools
    • Chrome (web)            • chrome             • web-javascript • Google Chrome 84.0.4147.89

• No issues found!
```
</details>

**Comment by darshankawar** [comment]
Verified this on latest stable version using code sample provided and I don't see the issue anymore. I was unable to place / position the cursor in between emoji character running on web and mobile.

<details>
<summary> stable flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 3.3.7, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.3.7 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e99c9c7cd9 (2 days ago), 2022-11-01 16:59:00 -0700
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

Closing this as fixed.
